client/my-sites/patterns/components/pattern-library/index.tsx

```diff
@@ -73,34 +72,19 @@ function getPatternPermalink(
 }
 
 type PatternLibraryProps = {
-	category: string;
 	categoryGallery: CategoryGalleryFC;
-	isGridView?: boolean;
 	patternGallery: PatternGalleryFC;
-	patternTypeFilter: PatternTypeFilter;
-	referrer?: string;
-	searchTerm?: string;
 };
 
 export const PatternLibrary = ( {
-	category,
 	categoryGallery: CategoryGallery,
-	isGridView,
 	patternGallery: PatternGallery,
-	patternTypeFilter,
-	referrer,
-	searchTerm: urlQuerySearchTerm = '',
 }: PatternLibraryProps ) => {
 	const locale = useLocale();
 	const translate = useTranslate();
-
-	// Helps prevent resetting the search input if a search term was provided through the URL
-	const isInitialRender = useRef( true );
-	// Helps reset the search input when navigating between categories
-	const [ searchFormKey, setSearchFormKey ] = useState( category );
 	const navRef = useRef< HTMLDivElement >( null );
+	const { category, searchTerm, isGridView, patternTypeFilter, referrer } = usePatternsContext();
```

Comment:
I started integration of the Context with this root component, to prevent starting point of the drilling. I intentionally didn't refactored all the code to avoid extra effort during the review, it will be easy to do later, or maybe in context of the issues connected to a specific module. Anyway my main goal was - make introduction of the new approach and deprecate the old one.